src/react/components/pages/projectSettings/ProjectMetrics.tsx

```diff
@@ -0,0 +1,355 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import React, { Component } from "react";
+import _ from "lodash";
+import {
+    AssetState, IAssetMetadata, IProject, ITag,
+} from "../../../../models/applicationState";
+import { strings, interpolate } from "../../../../common/strings";
+import {
+    XYPlot, Sunburst, Hint, DiscreteColorLegend,
+    HorizontalGridLines, XAxis, YAxis, VerticalBarSeries, LabelSeries
+} from "react-vis";
+import "react-vis/dist/styles/radial-chart.scss";
+import "react-vis/dist/styles/plot.scss";
+import "./projectSettingsPage.scss";
+import { Spinner } from "@fluentui/react";
+
+/**
+ * Required properties for Project Metrics
+ * @member project - Current project to fill metrics table
+ */
+export interface IProjectMetricsProps {
+    project: IProject;
+}
+
+export interface IProjectMetricsState {
+    loading: boolean;
+    hoveredCell: any;
+    projectAssetsMetadata: IAssetMetadata[];
+    metricsContainerWidth: number;
+}
+
+/**
+ * @name - Project Form
+ * @description -
+ */
+export default class ProjectMetrics extends Component<IProjectMetricsProps, IProjectMetricsState> {
+    public state = {
+        loading: true,
+        hoveredCell: null,
+        sourceDocuments: [],
+        projectAssetsMetadata: [],
```

Comment:
Isn't it better to initialize here, instead of in didMount so we don't set state twice?